### PR TITLE
Add Vars

### PR DESCRIPTION
**Pull request by m1**

### CODE REVIEW

**Comment by marcuschaplais** [comment]
:+1:

**Comment by jadb** [comment]
@m1, please, for the last time, read the guidelines.

FWIW, this package:

- is not widely recommended, well-known, fill a niche nor unique.
- is submitted by it's author even though it doesn't mean the criteria above
- has a typo in submission

I am closing and will *not* frown-upon future submissions (as suggested in the guidelines), but for now it's a no.

**Comment by m1** [comment]
@jadb lol sigh. Right, for the first point; It has more downloads on packagist than these previous merges; https://github.com/ziadoz/awesome-php/pull/556 - https://github.com/ziadoz/awesome-php/pull/529 -- As for niche or unique, explain? It's kind of meant as a lightweight alternative to zend-config, that's kind of the point of the library.

Also where's the typo?

**Comment by marcuschaplais** [comment]
@jadb Ditto @m1's point about it being pretty popular, we use it on all our client sites and the openplatform api which gets thousands of hits a day, it's super handy and less fuss than using something like `symfony/config`

@m1 I think the typo he was talking about was you're missing a comma between PHP and includes

File: README.md [hunk 1]
```diff
@@ -136,6 +136,7 @@ Please see [CONTRIBUTING](https://github.com/ziadoz/awesome-php/blob/master/CONT
 *Libraries for loading configuration files*
 
 * [Zend-config](https://github.com/zendframework/zend-config) - Config component from Zend Framework
+* [Vars](https://github.com/m1/Vars) - Vars is a configuration file loader for PHP includes Silex provder.
```
Comment (jadb) [inline README.md]: Typo: `provder`.

**Comment by jadb** [comment]
@m1, this is going no where. 

If @ziadoz or @josegonzalez wish to re-open and include it, I will not oppose it (nothing personal against you), but as far as I am concerned, the guidelines are there for a reason and IMO, this submission doesn't meet them.

**Comment by marcuschaplais** [comment]
@jadb You didn't really answer any of his questions/points, or even explain why it was closed

**Comment by m1** [comment]
@marcuschaplais I think I have to open up a new pull request because of a typo :disappointed:

**Comment by jadb** [comment]
@m1 no need, it could be pushed on top of this one, but that's beside the point.

@marcuschaplais I don't see why I should continue the argument, I gave my comment and if anyone of the other curators would like to revert my decision, they are free to do so.

**Comment by marcuschaplais** [comment]
@jadb It's not an argument? He was just asking you to explain yourself in terms of the first point he made.

**Comment by m1** [comment]
@marcuschaplais This has all gone a bit weird. I think I made him mad over this: https://github.com/ziadoz/awesome-php/pull/560#issuecomment-177265048 which was just a comment on the guidelines for packages. I think this repo needs some impartial contributors tbh.

**Comment by jadb** [comment]
@m1, am not mad and am being really impartial. I re-iterate, if others think it should be reverted, I won't oppose it nor defend my case more than I already did. 

That was my honest and impartial opinion,  but you, *understandably so*, can't be unbiased (probably why you are feeling it has gone all weird, while I find it normal).

**Comment by m1** [comment]
@jadb But the thing is, I am being unbiased.

I don't mind if my pull req gets denied for good reason and it gets explained -- it's just that you didn't explain why it got closed

**Comment by jadb** [comment]
@m1 but I did. You just don't want to understand my point of view and I don't feel like I should (nor would like to make it a norm here) go into details. 

The guidelines I contribute time by are clear and I could have just closed this with no explanation (maybe what I should have done not to stir this long time-consuming conversation), based on it being submitted by its author and not meeting criteria. 

Please, just let @ziadoz  or @josegonzalez check and if they find it should make it, they will revert. Arguing over it won't change the current outcome.

**Comment by marcuschaplais** [comment]
@jadb The problem @m1 has is that you didn't explain yourself re: 

>  Right, for the first point; It has more downloads on packagist than these previous merges; #556 - #529 (both merged by you)
> As for niche or unique, explain? It's kind of meant as a lightweight alternative to zend-config, that's kind of the point of the library.

**Comment by jadb** [comment]
@marcuschaplais I repeat, I will not go into details. I don't have to nor do I think it is fruitful. 

Maybe better ask him why the many he had initially submitted were not submitted instead? They are definitely more popular/recommended authors/vendors in the PHP community as well. Or maybe the PR was all started to add 2 personal packages to the list?

**Comment by m1** [comment]
@jadb Or maybe I was just working my way down the original list I submitted. Think you're being a bit paranoid and petty tbh.

**Comment by jadb** [comment]
@m1 - weird, the zend one you started the new PR serie with was not on the original list you claim to be working your way down.

**Comment by m1** [comment]
@jadb lol maybe I thought of a new package to add maybe?

**Comment by marcuschaplais** [comment]
@jadb Then why even bother having this project on github at all? Github should be for community projects, with a community input, that's what open source is about.

**Comment by jadb** [comment]
@m1, well maybe you should have started with the more popular/recommended ones in the PHP community. After all, this is the 'awesome' list not product hunt.

@marcuschaplais - thanks for all your input.

Sorry but I will not be taking part in this discussion anymore.

**Comment by marcuschaplais** [comment]
@jadb FYI `Vars` fit my bill but "most popular", "recommended" ones such as `symfony/yaml` did not. Vars is more lightweight and supports custom loaders.

**Pull request closed without merging**